### PR TITLE
Move license checks back to BuildPlugin

### PR DESCRIPTION
**Pull request by jakelandis**
~Some standalone projects such as :distribution:archives:integ-test-zip have Java code 
via tests, but are not need license and notice files. Applying the new `ElasticsearchJavaPlugin`
in favor over the elder `Standalone*Plugin` can result in these non-jar
projects test projects throwing an error due to the lack of the license or notice file.~ 

This commit moves the license check out of the new `ElasticsearchJavaPlugin` and applies
only via the elder `BuildPlugin`

related: #56841

======

~I'm not 100% this is the right direction, but seems pretty harmless to allow projects to opt out of the default jar creation.~ This PR is related to https://github.com/elastic/elasticsearch/pull/56841#discussion_r426902049 ...trying to move away from the conditional checks to that ensure specific plugins are applied.

### CODE REVIEW

**Comment by rjernst** [comment]
To me this seems like something that should go in the publish plugin, since that is where we need the jars to contain the license files. However, we don't apply the publish plugin to all projects that go into our distributions (eg under libs). I think there is an argument for another plugin, one for any  project that will end up in the distribution but is not published. I can't think of what to call it right now though..

**Comment by jakelandis** [comment]
@rjernst I don't disagree, however would you mind if I logged a follow up issue to address that and proceed with this shorter term solution here ? The change here helps to unblock another PR.

**Comment by mark-vieira** [comment]
> However, we don't apply the publish plugin to all projects that go into our distributions (eg under libs).

Actually we _do_ since we apply the `ElasticsearchBuild` plugin to those, which in turn applies the publish plugin.

I'd move the jar license/notice check into `ElasticsearchBuild` plugin. That plugin might change evolve but it's the best place now as that's applied to stuff we "assemble". I anticipate going forward we'll remove the publish plugin from that, and explicitly apply publishing to specific modules. That way we can apply `elasticsearch.build` to `:libs` projects, w/o all the publishing stuff (generate POM) but the build plugin still applies conventions, precommit checks, etc that aren't neccessarily applicable to standalone qa projects (i.e. elasticsearch.java).

**Comment by mark-vieira** [comment]
@jakelandis discussed with @rjernst and we've agreed to just move this into `BuildPlugin` instead of `ElasticsearchJavaPlugin`. We can ditch all the `enabled` stuff and this still accomplishes what you want, which is the ability to use `ElasticsearchJavaPlugin` w/o dealing with these strict enforcements.

**Comment by jakelandis** [comment]
thanks @rjernst @mark-vieira - I reverted the original commit(s) and moved the method call out of `ElasticsearchJavaPlugin` and into `BuildPlugin`.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy [hunk 1]
```diff
@@ -82,6 +82,7 @@ class BuildPlugin implements Plugin<Project> {
             )
         }
         project.pluginManager.apply('elasticsearch.java')
+        ElasticsearchJavaPlugin.configureJars(project)
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy:85]: We shouldn't be calling these methods statically; that is what we are trying to move away from. Can you please move the notice/license configuration to a new method in BuildPlugin, and leave the other stuff intact?

Comment (jakelandis) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy:85]: I can, however, aren't we also moving away from adding code to the Groovy parts of the build ?  (I can move the code to Groovy and keep in the code in idiomatic Java, but isn't it still is going the wrong way ?) 

> We shouldn't be calling these methods statically

In that case, should these methods not be `public static` ?

Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy:85]: I will be fishing moving this code to java soon. I'll take care of re-javaizing it.

Comment (jakelandis) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: OK ... I pulled the groovy version from git history and removed the Java version. https://github.com/elastic/elasticsearch/pull/56975/commits/908ee17737f1a283cf0535d0bfe102fa65144fb4

Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: Sorry that isn't what I meant. *Only* the notice/license code should be moved. That is the problem you are running into right?

Comment (jakelandis) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: pulled only the license and notice parts back to the groovy plugin

**Review by mark-vieira — APPROVED** [review]
Changes LGTM 👍

**Pull request merged**